src/sherpa_ai/config/task_config.py

```diff
@@ -0,0 +1,41 @@
+import re
+from argparse import ArgumentParser
+from typing import List, Optional, Tuple
+
+from pydantic import BaseModel
+
+
+class AgentConfig(BaseModel):
+    verbose: bool = False
+    gsite: Optional[str] = None
+    do_reflect: bool = False
+
+    @classmethod
+    def from_input(cls, input_str: str) -> Tuple[str, "AgentConfig"]:
+        """
+        parse input string into AgentConfig. The configurations are
+        at the end of the string
+        """
+        parts = re.split(r"(?=--)", input_str)
+        configs = []
+
+        for part in parts[1:]:
+            part = part.strip()
+            configs.extend(part.split())
+
+        return parts[0].strip(), cls.from_config(configs)
+
+    @classmethod
+    def from_config(cls, configs: List[str]) -> "AgentConfig":
+        parser = ArgumentParser()
+
+        parser.add_argument("--verbose", action="store_true")
```

Comment:
suggestion: add help messages for these arguments so users know what the args do

Reply:
I think the help messages can be helpful. However, these help messages will not be directly shown to the user unless they are running Sherpa from the commandline. Let's add the helper message first and then find a way to show the help menu in general.